app/models/comment.rb

```diff
@@ -49,14 +49,13 @@ def notify(action)
   def mentions
     @mentions = nil if content_changed?
     @mentions ||= begin
-      mentioned_users = []
-      HTML::Pipeline::MentionFilter.mentioned_logins_in(content, /[a-z0-9][a-z0-9\-@\.]*/) do |match, login, is_mentioned|
-        if (mentioned_user = User.find_by_email(login))
-          mentioned_users << mentioned_user
-        end
+      emails = []
+      HTML::Pipeline::MentionFilter.mentioned_logins_in(content, /[a-z0-9][a-z0-9\-@\.]*/) do |_, login, _|
+        emails << login
       end
 
-      mentioned_users
+      project = commentable.project
+      project.authors.where(email: emails.uniq)
```

Comment:
Nice refactor 🙌

Although the `Evidence` end of it gives me an allergic reaction. Is it just me?

Reply:
The `Evidence` delegate brings it together some. I think it was the only `item` without it so now they're more consistent at least.